TTD/LiveRecorderApiSample/main.cpp

```diff
@@ -0,0 +1,190 @@
+    if (ttdResult <= 32)
+    {
+        DWORD extendedError = GetLastError();
+        if (ttdResult == SE_ERR_ACCESSDENIED && extendedError == ERROR_CANCELLED)
+        {
+            std::cerr << "Elevation request was denied by the user so we can't proceed.\n";
+        }
+        else
+        {
+            std::cerr << std::format("Couldn't run TTD.exe. Error: {}, extended error: {}\n", ttdResult, extendedError);
+        }
+        return 1;
+    }
+
+    if (WaitForSingleObject(recordingStartedEvent, INFINITE) != WAIT_OBJECT_0)
+    {
+        std::cerr << "Error in the wait\n";
+        return 1;
+    }
+
+    // TTD is now installed in the process and ready to record.
+    Microsoft::WRL::ComPtr<TTD::ILiveRecorder> const pRecorder{ TTD::MakeLiveRecorder(ClientGuid, "Hello, there!") };
+    if (pRecorder == nullptr)
+    {
+        std::cerr << "Failed to get the recorder interface!\n";
+        return 2;
+    }
+
+    // We started TTD in manual mode, so nothing should be recording at this point.
+    if (auto const state = pRecorder->GetState(); state != TTD::ThreadRecordingState::NotStarted)
+    {
+        std::cerr << std::format("main: The current thread should not be recording yet! State is {}\n", GetStateName(state));
+        return 2;
+    }
+
+    // Record a simple single-threaded sort algorithm.
+    // The sorting is recorded as a single island with its own activity ID.
+    auto stdSortBuffer = randomSequence;
+    pRecorder->StartRecordingCurrentThread(StdSortActivity, TTD::InstructionCount::Max);
+    std::ranges::sort(stdSortBuffer);
+    pRecorder->StopRecordingCurrentThread();
+
+    // Verify that the sort worked as expected while being recorded.
+    if (!std::ranges::is_sorted(stdSortBuffer))
+    {
+        std::cerr << "main: std::sort didn't work right.\n";
+        return 2;
+    }
+
+    // Record all the pieces of a simple multithreaded sort algorithm.
+    // The sorting of all the asynchronous pieces is recorded in the trace,
+    // as islands belonging to the same activity.
+    auto multithreadedSortBuffer = randomSequence;
+    pRecorder->StartRecordingCurrentThread(MultithreadedSortActivity, TTD::InstructionCount::Max);
+    MultithreadedSort(pRecorder.Get(), multithreadedSortBuffer);
+    pRecorder->StopRecordingCurrentThread();
+
+    // Verify that the multithreaded sort worked as expected while being recorded.
+    if (!std::ranges::is_sorted(multithreadedSortBuffer))
+    {
+        std::cerr << "main: Multithreaded sort didn't work right.\n";
+        return 2;
+    }
+
+    // Obtain and print the full path to the TTD trace file that is being recorded.
+    wchar_t traceFilename[MAX_PATH];
+    pRecorder->GetFileName(traceFilename, std::size(traceFilename));
+    traceFilename[MAX_PATH - 1] = L'\0';
+    std::wcout << std::format(L"Trace file: {}\n", traceFilename);
+
+    // Close the recorder API, supplying a user data (in this case, just a short string).
+    pRecorder->Close("Adios!");
```

Comment:
>("Adios!" [](http://example.com/codeflow?start=20&length=9)

nit: I don't know if "Adios!" should be in the sample, just because some people may make assumptions and take offense.

Reply:
Right, who knows.